File: backend/src/models/Category.ts
import mongoose, { Document, Schema } from "mongoose";

export interface ICategory extends Document {
  title: string;
  color: string;
  icon?: string;
}

const CategorySchema: Schema = new Schema({
  title: { type: String, required: true },
  color: { type: String, required: true },
  icon: { type: String },
});

export default mongoose.model<ICategory>("Category", CategorySchema);

File: pylith/monolith/savvy/models/category.py
from django.db import models
from django.contrib.auth.models import User

class Category(models.Model):
    title = models.CharField(max_length=255)
    color = models.CharField(max_length=7)  # For hex colors like '#FFFFFF'
    icon = models.CharField(max_length=255, blank=True, null=True)
    
    user = models.ForeignKey(User, on_delete=models.CASCADE)

    def __str__(self):
        return self.title
